crypto/pkcs12/p12_init.c

```diff
@@ -42,3 +44,19 @@ PKCS12 *PKCS12_init(int mode)
     PKCS12_free(pkcs12);
     return NULL;
 }
+
+PKCS12 *PKCS12_init(int mode)
+{
+    return PKCS12_init_ex(mode, NULL, NULL);
+}
+
+void PKCS12_set_libctx(PKCS12 *p12, OSSL_LIB_CTX *ctx)
+{
+    p12->authsafes->ctx.libctx = ctx;
+}
+
+void PKCS12_set_propq(PKCS12 *p12, const char *propq)
+{
+    p12->authsafes->ctx.propq = (char *)propq;
```

Comment:
strdup

Reply:
I am not even sure we want the set_libctx and set_propq functions. The code that needs non-default context and/or propq should simply use the init_ex().

Reply:
I think these are probably only needed if the p12 object is created using d2i_PKCS12 but I'll have the check this. The alternative is to create a custom d2i_PKCS12_ex() which passes the libctx/propq.

Reply:
If you can create an empty PKCS12 object with the libctx, propq set then that can be passed to the d2i call.. That should be sufficient.. X509 does this. it is then up to the user to pass it in, instead of just using the returned object.

Reply:
Sure, that seems to work fine. I'll remove the set_* functions.